### PR TITLE
LP-1951 Add idea model and register in admin

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### Description

[LP-1951](https://philanthropyu.atlassian.net/browse/LP-1951) Creating idea model and registering with admin

### Notes

Unit test are included

### CODE REVIEW

File: openedx/features/idea/tests/factories.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        django_get_or_create = ('user', 'organization')
+
+    user = factory.SubFactory(UserFactory)
+    title = factory.Faker('pystr', min_chars=1, max_chars=TITLE_MAX_LENGTH)
+    overview = factory.Faker('pystr', min_chars=1, max_chars=OVERVIEW_MAX_LENGTH)
+    description = factory.Faker('word')
+    organization = factory.SubFactory(OrganizationFactory)
+    organization_mission = factory.Faker('word')
+    country = 'PK'
+    city = factory.Faker('pystr', min_chars=1, max_chars=CITY_MAX_LENGTH)
```
Comment (zeshanarifios) [inline openedx/features/idea/tests/factories.py:22]: How about making factories separately for location and organization base and inherit IdeaFactory from them?

File: openedx/features/idea/models.py [hunk 2]
```diff
@@ -2,5 +2,69 @@
...
+    organization = models.ForeignKey(
+        Organization,
+        related_name='%(app_label)s_%(class)ss',
+        related_query_name='%(app_label)s_%(class)s',
+        on_delete=models.CASCADE
+    )
+    organization_mission = models.TextField()
+
+    class Meta:
+        abstract = True
```
Comment (zeshanarifios) [inline openedx/features/idea/models.py:56]: these classes should be moved to some common place. 
Let's please discuss with @ahmed-faraz46 @saad-waqar for suitable place.

File: openedx/features/idea/admin.py [hunk 3]
```diff
@@ -3,4 +3,15 @@
...
-# Register your models here.
+from .models import Idea
+
+
+class IdeaAdmin(admin.ModelAdmin):
+    """Django admin customizations for Idea model."""
+
+    list_display = ('title', 'user', 'organization', 'city', 'country')
+    search_fields = ('user__username', 'organization__label')
+    raw_id_fields = ('user',)
```
Comment (zeshanarifios) [inline openedx/features/idea/admin.py:14]: Great 👍

File: openedx/features/idea/models.py (hunk 2 above)
Comment (zahid-arbisoft) [reply openedx/features/idea/models.py:56]: Lets discuss this and defer it until marketplace

**Pull request merged**